internal/config/config.go

```diff
@@ -128,6 +151,47 @@ func (c *Config) getServerConfig(config []byte) (*ServerConfig, error) {
 	}, nil
 }
 
+func (c *Config) getInstallationConfigs() *InstallationConfigs {
+	return &InstallationConfigs{
+		cache: gc.New(1*time.Hour, 10*time.Minute),
+	}
+}
+
+// GetConfig retreives and stores the configuration for a repository
+func (ic *InstallationConfigs) GetConfig(ctx context.Context, repo *github.Repository, sha string, client *github.Client) *InstallationConfig {
+	repoID := strconv.FormatInt(repo.GetID(), 10)
+
+	if config, found := ic.cache.Get(repoID); found {
+		return config.(*InstallationConfig)
+	}
+
+	config := InstallationConfig{}
+	var rawConfig string
+
+	fc, _, resp, err := client.Repositories.GetContents(
+		ctx,
+		repo.GetOwner().GetLogin(),
+		repo.GetName(),
+		installationConfigFileLocation,
+		&github.RepositoryContentGetOptions{Ref: sha},
```

Comment:
Note: this means that the PR will run against settings relative to its SHA _not_ what is in master. This is so that updates to the rules can be seen on the PR that includes them, though this also means stale PRs will show stale positives, which might cause confusion...